docs/doc/60-contributing/03-rfcs/20220909-Idempotent-Copy.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Idempotent Copy
+description: Avoid duplicating when copy stage files into a table
+---
+
+- Tracking Issue: https://github.com/datafuselabs/databend/issues/6338
+
+## Summary
+
+When streaming copy stage files into a table, there is a chance that some files have already been copied, So it needs some ways to avoid duplicate copying files, make it an `idempotent` operation.
+
+## Save copy into table stage files meta information in meta service
+
+Whenever copy stage files into a table, save the stage file meta information into the meta service:
+
+- key: combined with `(tenant, database, table, file name)`.
+- value: value MUST includes all the meta of a stage file, such as `content-length`,`etag`,`last modified`.
+
+
+
+![](/img/rfc/20220909-Idempotent-Copy/stage-file-meta.png)
```

Comment:
Please use `/img/rfc/20220909-idempotent-copy/stage-file-meta.png`

So that we don't have a upcase folder.